### DISCUSSION TITLE
QCefView对于CEF版本选择问题

### DISCUSSION THREAD

**Discussion by lightexpensive** [CefView/QCefView#203] [Q&A] [answered]
想了解为什么把cef版本从95.7.12升到102.0.10？
是因为要支持新特性吗？或者是为了跟随官方版本升级修复一些已知CEF bug？或者是什么其它原因呢？
另外为什么选择的是102.0.10，是有做过调研吗？
想知道是否有考虑基于某个稳定版本进行维护开发，然后从CEF或Chromium官方pick一些问题修复或新特性支持。
众所周知，**CEF已经变得越来越庞大**，这对一些开发者或企业来说将来或许会是个瓶颈问题；
另外，有消息称**Chromiu 110后的版本不再支持Win 7系统**，关于跟随官方版本迭代，这也是个值得考虑的问题；
本人有测验过不同平台下的Chrome 、Chromium、CEF官方demo，发现同一个Chromium版本中，Chrome一般表现良好，Chromium有些版本的中文输入预选框在某些情况（窗口弹出页面加载后，web输入框自动取得焦点时）会存在偏移问题，部分CEF版本在中文输入预选框甚至中文输入支持，在某些情况下无法正常支持，另外CEF linux对中文及web文件选择框（html input file）不支持。
因为国产化逐渐得到重视，所以linux平台支持的问题应该也是值得投入关注的。

**Comment by tishion** [comment, marked as answer]
你提的这些问题都很好。
- 关于版本问题
开发者可以自由选择CEF的版本，但是因为CEF的接口有变更，所以可能需要自行适配，从100版本开始QCefView已经改变为通过CEF的版本来控制这些变更的，但是更旧的版本没有支持，你可以直接通过commit history切换到依赖旧版本的提交。

- 关于一些特性支持问题例如输入
  这个是CEF自身的问题，你经过调研也发现Chrome支持最好，那你要考虑Chrome是一个全球顶级的公司内的一个大规模团队开发维护的。

- 关于Linux平台支持的问题
 这也是CEF自身的问题，支持这些是需要精力和资源投入去研发的。


QCefView对我个人来说只是一个业余项目，开源也只是带着纯粹的技术分享的目的，并没有从商业或者民族大义这些角度去做思考或者规划，如果您有很多很好的想法，随时欢迎您为这个项目带来一些贡献和提升，十分感谢！
